src/ServerMain.cpp

```diff
@@ -147,6 +147,28 @@ int main(int argc, char** argv) {
       optionFactory.getProgramOption<"enable-prefilter-on-index-scans">(),
       "If set to false, the prefilter procedures for FILTER expressions are "
       "disabled.");
+  add("group-by-hash-map-enabled",
+      optionFactory.getProgramOption<"group-by-hash-map-enabled">(),
+      "If set, then the GROUP BY operation will use a hash map to store "
+      "intermediate results. This is faster for small groups, but uses more "
+      "memory. If not set, the GROUP BY operation will use a sort-based "
+      "algorithm.");
+  add("group-by-sample-enabled",
+      optionFactory.getProgramOption<"group-by-sample-enabled">(),
+      "If set, then the GROUP BY operation will use a sampling-based hybrid "
+      "algorithm to decide whether to use a hash map or a sort-based algorithm "
+      "for grouping. If not set, the GROUP BY operation will always use a "
+      "sort-based algorithm.");
```

Comment:
In the final version this probably is an enum
(never-hash-map/always-hash-map/hash-map-via-sample-threshold.